app/models/measurement_list_reader.rb

```diff
@@ -18,74 +18,39 @@ def load
 
     @measurements = Measurement.where(mcc_filter: [nil, mcc])
     @measurements = filter_by_show_hide
-    # it might be nice to run this each separate threads since they do 3 GR calls each
-    # Spencer suggests waiting until we test how the performs
-    # https://github.com/tra/spawnling might be a good option, or a vanilla thread pool
-    # https://blog.engineyard.com/2014/ruby-thread-pool
-    @measurements.each(&method(:load_gr_value))
+    load_measurements_from_gr
     split_children
   end
 
   private
 
-  def load_gr_value(measurement)
-    return load_historic_gr_values(measurement) if historical
-    levels = Power.current.try(:measurement_levels) || [:total, :local, :person]
-    levels.each do |level|
-      measurement_level_id = measurement.send("#{level}_id")
-      filter_params = gr_request_params(level)
-      gr_resp = GlobalRegistryClient.client(:measurement_type).find(measurement_level_id, filter_params)
-      value = gr_resp['measurement_type']['measurements'].map { |m| m['value'].to_f }.sum
-      measurement.send("#{level}=", value)
+  def load_measurements_from_gr
+    work_q = Queue.new
+    params = gr_loading_params
+    @measurements.each { |m| work_q.push m }
+    workers = (1..ENV['MEASUREMENT_THREAD_COUNT'].to_i).map do
+      Thread.new do
```

Comment:
Here's another thing I just thought of: `GlobalRegistryClient` uses a thread variable to store the global registry params, but here you're using multiple threads which I think could mess that up for the case of a systems measurement endpoint where we are supposed to load the measurement with a specific global registry key for the system (assuming it's supposed to work like that).

What would you think about passing in the global registry client directly to your `Measurement::GrValueLoader` class so that it's dependency is clear?